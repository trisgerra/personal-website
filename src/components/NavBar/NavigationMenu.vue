<template>
  <div :class="this.navBarMenuClass">
    <div class="navbar-start">
      <router-link
        v-for="route in routes"
        v-bind:key="route.name"
        class="menu-item"
        :to="route.path">
        <a class="navbar-item navbar-router">{{route.name}}</a>
      </router-link>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import { IRoute, routes } from '../../router';

@Component
export default class NavigationMenu extends Vue {
  @Prop() isActive!: Function;

  routes: IRoute[] = routes;

  get navBarMenuClass(): string {
    return this.isActive ? 'navbar-menu is-active' : 'navbar-menu';
  }
}
</script>

<style scoped lang="scss">
@import '~bulma/sass/utilities/mixins';

.navbar-router {
  background-color: white;
  font-weight: 400;
  height: 100%;
  color: #151616;
  &:hover {
    color: #151616;
    font-weight: 600;
  }
}

.router-link-exact-active {
  a {
    background-color: white;
    font-weight: 600;
  }
}
@include desktop {
  .navbar-router {
      border-bottom: 3px solid white;
      &:hover {
        border-bottom: 3px solid #151616;
      }
    }
    .router-link-exact-active {
        border-bottom: 3px solid #151616;
    }
}

@include touch {
  .navbar-router {
      border-left: 2px solid white;
      &:hover {
        border-left: 2px solid #151616;
      }
    }
}
</style>
