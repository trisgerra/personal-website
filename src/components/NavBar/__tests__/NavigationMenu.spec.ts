import { Wrapper } from '@vue/test-utils';
import NavigationMenu from '../NavBar.vue';
import createWrapper from '@/utils/CreateWrapper';

describe('NavigationMenu', () => {
  it('should render as expected', () => {
    const wrapper: Wrapper<NavigationMenu> = createWrapper(NavigationMenu);
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.is('.navbar-brand'));
  });

  it('should render as expected', () => {
    const wrapper: Wrapper<NavigationMenu> = createWrapper(NavigationMenu);
    expect(wrapper.findAll('.menu-item').length).toBe(5);
  });
});
