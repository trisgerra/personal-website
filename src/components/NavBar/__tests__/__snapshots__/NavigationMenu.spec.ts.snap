// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMenu should render as expected 1`] = `
<nav
  aria-label="main navigation"
  class="navbar"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <a
      class="menu-item router-link-exact-active router-link-active"
      href="#/"
    >
      <div
        class="navbar-item navbar-logo"
      >
        <img
          alt="ADG Official"
          height="38"
          src="../../assets/logo.png"
        />
      </div>
    </a>
     
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger"
      role="button"
    >
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
    </a>
  </div>
   
  <div
    class="navbar-menu"
  >
    <div
      class="navbar-start"
    >
      <a
        class="menu-item router-link-exact-active router-link-active"
        href="#/"
      >
        <a
          class="navbar-item navbar-router"
        >
          Home
        </a>
      </a>
      <a
        class="menu-item"
        href="#/blog"
      >
        <a
          class="navbar-item navbar-router"
        >
          Blog
        </a>
      </a>
      <a
        class="menu-item"
        href="#/design"
      >
        <a
          class="navbar-item navbar-router"
        >
          Design
        </a>
      </a>
      <a
        class="menu-item"
        href="#/projects"
      >
        <a
          class="navbar-item navbar-router"
        >
          Projects
        </a>
      </a>
    </div>
  </div>
</nav>
`;
